### PR TITLE
feature: start new module for retrieving prebuilt SageMaker image URIs

### PR DESCRIPTION
**Pull request by laurenyu**
*Issue #, if available:*
#1464

*Description of changes:*
This commit also introduces configuration for Chainer URIs. I started with Chainer because it's the simplest, but I'll probably pick something a bit more complicated (e.g. TF) next.

I didn't actually replace the logic in the Chainer classes because they just inherit from the Framework classes. I also didn't replace the integ test fixtures because I plan on renaming them in the process, and that will be a large change on its own.

*Testing done:*
`tox --parallel all tests/unit`

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

#### General

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-python-sdk/blob/master/CONTRIBUTING.md#committing-your-change)
- [ ] I have passed the region in to all S3 and STS clients that I've initialized as part of this change.
- [x] I have updated any necessary documentation, including [READMEs](https://github.com/aws/sagemaker-python-sdk/blob/master/README.rst) and [API docs](https://github.com/aws/sagemaker-python-sdk/tree/master/doc) (if appropriate)

#### Tests

- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [x] I have checked that my tests are not configured for a specific region or account (if appropriate)
- [ ] I have used [`unique_name_from_base`](https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/utils.py#L77) to create resource names in integ tests (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/sagemaker/image_uris.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    )
+
+
+def _registry_from_region(region, registry_dict):
+    """Returns the ECR registry (AWS account number) for the given region."""
+    available_regions = registry_dict.keys()
+    if region not in available_regions:
+        raise ValueError(
+            "Unsupported region: {}. "
+            "You may need to upgrade your SDK version (pip install -U sagemaker) newer regions. "
```
Comment (ajaykarpur) [inline src/sagemaker/image_uris.py:89]:
```suggestion
            "You may need to upgrade your SDK version (pip install -U sagemaker) for newer regions. "
```

File: src/sagemaker/image_uris.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        "Supported type(s): {}.".format(processor, instance_type, ", ".join(available_processors))
+    )
+
+
+def _validate_py_version(py_version, available_versions, framework, fw_version):
+    """Checks if the Python version is one of the supported versions."""
+    if py_version not in available_versions:
+        raise ValueError(
+            "Unsupported Python version for {} {}: {}. "
+            "You may need to upgrade your SDK version (pip install -U sagemaker) newer versions. "
```
Comment (ajaykarpur) [inline src/sagemaker/image_uris.py:123]:
```suggestion
            "You may need to upgrade your SDK version (pip install -U sagemaker) for newer versions. "
```

File: tests/conftest.py [hunk 3]
```diff
@@ -405,3 +400,12 @@ def pytest_generate_tests(metafunc):
         ):
             params.append("ml.p2.xlarge")
         metafunc.parametrize("instance_type", params, scope="session")
+
+    for fw in ("chainer",):
```
Comment (ajaykarpur) [inline tests/conftest.py:404]: Is the tuple with a single string intentional? Do you plan to add other frameworks here later?

Comment (laurenyu) [reply tests/conftest.py:404]: yep, the idea is to add all the frameworks here and remove their respective individually-defined fixtures

**Pull request merged**